ra2ce/analyses/indirect/origin_closest_destination.py

```diff
@@ -494,8 +494,10 @@ def find_closest_location(
         optimal_routes = []
         list_disrupted_destinations = []
         list_no_path = []
+        node_checked_has_path = dict()
```

Comment:
You can reverse a bit the logic here, to make it 'easier' using [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict) later on:
```suggestion
        node_checked_has_path = defaultdict(list)
```
Then in line 570 you can simply do:
```python
node_checked_has_path[ndat[self.od_key]).append(n)
```
Either way, I'm a bit in doubt of this functionality, perhaps it's good if we have a look at it together as it's not ideal having to modify a parameter and later returning it for the next iteration.

Reply:
OK, for now updated accordingly.